src/config/config-types.ts

```diff
@@ -126,6 +127,7 @@ export interface YamlBaseModel {
     groups?: string[];
     fieldGroups?: FieldGroupItem[];
     fields?: Field[];
+    presets?: any;
```

Comment:
Do we need `presets` on `YamlBaseModel`?
YamlBaseModel defines the structure of the actual yaml model file definition.
The Config interface defines the structure of the config object returned by the SDK with resolved presets.
From the code below, I see that you add presets to the variable of `Config` type only.

Reply:
A model will have presets as well (preset id's), right?
Would `BaseModel` be better?

Reply:
oh oh, sorry I confused some stuff myself...

So, the final `Config` type has `presets` which is a map of all the presets to their IDs:
https://github.com/stackbit/stackbit-sdk/blob/cbec02081449429959cdfdc344d5a1ad07bec2fe/src/config/config-types.ts#L6
👍

Then, every final model also has `presets` which is the array of preset IDs of a specific model.
You defined it inside `YamlBaseModel`:
https://github.com/stackbit/stackbit-sdk/blob/cbec02081449429959cdfdc344d5a1ad07bec2fe/src/config/config-types.ts#L130
This is OK and will not cause any problems, it is just it is not strict.

In any case, FYI...

There are two main Model types:
1. YamlModel - type of models as they should be defined in model files (and used to type Joi schemas)
https://github.com/stackbit/stackbit-sdk/blob/cbec02081449429959cdfdc344d5a1ad07bec2fe/src/config/config-types.ts#L113

2. Model - type of models as returned by `loadConfig()`
https://github.com/stackbit/stackbit-sdk/blob/cbec02081449429959cdfdc344d5a1ad07bec2fe/src/config/config-types.ts#L100

The "Model" basically extends the "YamlModel" plus adds the "name" field (in that "BaseModel" interface you noted). Although technically speaking, it is not true anymore :). Because separate model files do have `name` field, but we normalize the loaded files before validating them and convert all models into a map as if all models were defined in stackbit.yaml with model names as keys
But the "presets" are always added to the final result, so yeah, the ideal place for them will be the "BaseModel".